File: EasyIP.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.*;
import java.io.*;
public class EasyIP{
	public static void main(String[] args){
		GUI g = new GUI();
		try{
		g.initial();
		}catch(Exception e){System.out.print(e);};
	}
}
class GUI implements ActionListener,ItemListener{
	int selected;
	String data[][] = new String[10][5];
	JButton b1 = new JButton("添加");
	JButton b2 = new JButton("设置本地连接");
	JButton b3 = new JButton("设置无线连接");
	JComboBox cb1 = new JComboBox();
	JTextField tf1 = new JTextField(15);
	JTextField tf2 = new JTextField(15);
	JTextField tf3 = new JTextField(15);
	JTextField tf4 = new JTextField(15);
	GUI(){
		JFrame frame = new JFrame();
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		frame.setLayout(new BorderLayout());
		frame.setSize(600,480);
		JPanel panel1 = new JPanel();
		panel1.setPreferredSize(new Dimension(400,480));
		cb1.addItemListener(this);
		panel1.setLayout(new GridLayout(5,1));
		panel1.add(cb1);
		JPanel p1 = new JPanel();
		JLabel l1 = new JLabel("IP地址");
		p1.add(l1);
		p1.add(tf1);
		panel1.add(p1);
		JPanel p2 = new JPanel();
		JLabel l2 = new JLabel("网关");
		p2.add(l2);
		p2.add(tf2);
		panel1.add(p2);
		JPanel p3 = new JPanel();
		JLabel l3 = new JLabel("DNS1");
		p3.add(l3);
		p3.add(tf3);
		panel1.add(p3);
		JPanel p4 = new JPanel();
		JLabel l4 = new JLabel("DNS2");
		p4.add(l4);
		p4.add(tf4);
		panel1.add(p4);
		frame.add(panel1,BorderLayout.CENTER);
		JPanel panel2 = new JPanel();
		panel2.setPreferredSize(new Dimension(200,480));
		panel2.setLayout(new GridLayout(3,1));
		panel2.add(b1);
		panel2.add(b2);
		panel2.add(b3);
		frame.add(panel2,BorderLayout.EAST);
		frame.setVisible(true);
	}
	public void actionPerformed(ActionEvent e)
	{
		String net = new String();
		if (e.getSource()==b2)
			net = "Local Area Connection";
		if (e.getSource()==b3)
			net = "Wireless Network Connection";
		if (e.getSource()!=b1){
			if(cb1.getSelectedItem()!="自动获取"){
				try{
				Runtime.getRuntime().exec("netsh interface ipv4 set address "+net+" static "+data[selected][1]+" 255.0.0.0 "+data[selected][2]);
				Runtime.getRuntime().exec("netsh interface ipv4 set dnsservers name="+net+" static "+data[selected][3]+" primary validate=no");
				Runtime.getRuntime().exec("netsh interface ipv4 add dnsservers "+net+" "+data[selected][4]+" validate=no");
				}
				catch(Exception ex){};
			}
			else{
				try{Runtime.getRuntime().exec("netsh interface ipv4 set address "+net+" source=dhcp");}catch(Exception ex){};
			}
		}
	}
	public void itemStateChanged(ItemEvent e){
		if(e.getStateChange()==ItemEvent.SELECTED){
			for(int i=0;i<10;i++){
				if(cb1.getSelectedItem()==data[i][0]){
					tf1.setText(data[i][1]);
					tf2.setText(data[i][2]);
					tf3.setText(data[i][3]);
					tf4.setText(data[i][4]);
					selected=i;
				}
				if(cb1.getSelectedItem()=="自动获取"){
					tf1.setText("");
					tf2.setText("");
					tf3.setText("");
					tf4.setText("");
				}
			}
		}
	}
	void initial () throws Exception
	{
		String s = new String();
		File file = new File("D:/EasyIP/EasyIP.txt");
		BufferedReader br = new BufferedReader(new FileReader(file));
		s = br.readLine();
		int flag = 0;
		int flag2 = 0;
		while(s!=null){
			data[flag2][flag]=s;
			if(flag==0)
				cb1.addItem(s);
			flag++;
			if(flag==5){
				flag=0;
				flag2++;
			}
			s=br.readLine();
		}
		cb1.addItem("自动获取");
	}
}

File: EasyIP.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define N 3
char *NAME[N],*IP[N],*GW[N],*YM[N],*DNS1[N],*DNS2[N],*NET[N],TEMP[1000];
void DELETE ( char *data )
{
     char *a;
     for ( a=data ; *a!= '\0' ; a++)
     {
         if ( *a == '\n' )
         *a = '\0';
     }
}
void EDIT()
{
	int i,a;
	FILE *add;
	putchar('\n');
	for ( i = 1 ; i <= N ; i++ )
		printf("%d.%s" , i , NAME[i-1] );
	putchar('\n');
	printf("1.Add\n2.Delete\n3.Open with Notepad\n0.Back to Menu\n");
	scanf("%d",&a);
	if ( a == 3 )
		system ("IP.txt");
	if ( a == 0 )
	{
		putchar ( '\n' );
		return;
	}
	EDIT();
}
void CLEAR ()
{
	system ("netsh interface ipv4 set address 本地连接 source=dhcp\nnetsh interface ipv4 set address 无线网络连接 source=dhcp" );
}
void PrintMenu()
{
	int i,choice;
	FILE *fp;
	if ( !( fp = fopen ( "IP.txt" , "r") ) ) exit(0);
	for ( i = 0 ; i < N ; i++ )
	{
		NAME[i] = (char*) malloc ( 20 * sizeof(char) );
		IP[i] = (char*) malloc ( 20 * sizeof(char) );
		GW[i] = (char*) malloc ( 20 * sizeof(char) );
		YM[i] = (char*) malloc ( 20 * sizeof(char) );
		DNS1[i] = (char*) malloc ( 20 * sizeof(char) );
		DNS2[i] = (char*) malloc ( 20 * sizeof(char) );
		NET[i] = (char*) malloc ( 20 * sizeof(char) );
		printf ( "%d:" , i+1 );
		fputs( fgets ( NAME[i] , 20 , fp ) , stdout );
		DELETE ( fgets ( IP[i] , 20 , fp ) );		
		DELETE ( fgets ( GW[i] , 20 , fp ) );
		DELETE ( fgets ( YM[i] , 20 , fp ) );
		DELETE ( fgets ( DNS1[i] , 20 , fp ) );
		DELETE ( fgets ( DNS2[i] , 20 , fp ) );
		DELETE ( fgets ( NET[i] , 20 , fp ) );
	}
	fclose ( fp );
	printf ( "%d:Auto\n%d:Edit\n0:Exit\n" , N+1 ,N+2 );
	scanf( "%d" , &choice );
	if ( choice == 0 ) 
		exit(0);
	if ( choice < N+1 )
	{
		strcpy ( TEMP , "netsh interface ipv4 set dnsservers name=" );
		strcat ( TEMP , NET[choice-1] );
		strcat ( TEMP , " static " );
		strcat ( TEMP , DNS1[choice-1] );
		strcat ( TEMP , " primary validate=no" );
		system(TEMP);
		strcpy ( TEMP , "netsh interface ipv4 add dnsservers ");
		strcat ( TEMP , NET[choice-1] );
		strcat ( TEMP , " ");
		strcat ( TEMP , DNS2[choice-1] );
		strcat ( TEMP , " validate=no\n" );
		system(TEMP);
		strcpy ( TEMP , "netsh interface ipv4 set address ");
		strcat ( TEMP , NET[choice-1] );
		strcat ( TEMP , " static ");
		strcat ( TEMP , IP[choice-1] );
		strcat ( TEMP , " " );
		strcat ( TEMP , YM[choice-1]  );
        	strcat ( TEMP , " " );
		strcat ( TEMP , GW[choice-1]  );
		CLEAR ();
		system(TEMP);
		exit ( 0 );
 	}
	if ( choice == N+1 )
		CLEAR ();
	if ( choice == N+2 )
		EDIT();
	PrintMenu();
}
int main()
{
	while(1)
		PrintMenu();
	return 0;
}
